config/cmake/HDFCompilerFlags.cmake

```diff
@@ -465,3 +465,33 @@ if (HDF5_ENABLE_OPTIMIZATION)
   list (APPEND HDF5_CMAKE_C_FLAGS "${OPTIMIZE_CFLAGS}")
 endif ()
 MARK_AS_ADVANCED (HDF5_ENABLE_OPTIMIZATION)
+
+#-----------------------------------------------------------------------------
+# The build mode flags are not added to CMAKE_C_FLAGS, so create a separate
+# variable for them so they can be written out to libhdf5.settings and
+# H5build_settings.c
+#-----------------------------------------------------------------------------
+set (HDF5_BUILD_MODE_C_FLAGS "")
+set (HDF5_BUILD_MODE_Fortran_FLAGS "")
```

Comment:
Shouldn't the Fortran and CXX flags be set in the HDFFortranCompilerFlags.cmake and HDFCXXCompilerFlags.cmake files?

Reply:
I can do that